reporters/kamon-prometheus/src/main/scala/kamon/prometheus/embeddedhttp/SunEmbeddedHttpServer.scala

```diff
@@ -32,24 +35,40 @@ class SunEmbeddedHttpServer(hostname: String, port: Int, scrapeSource: ScrapeSou
         if (httpExchange.getRequestURI.getPath == "/metrics") {
           val data = scrapeSource.scrapeData()
           val bytes = data.getBytes(StandardCharsets.UTF_8)
-          httpExchange.sendResponseHeaders(200, bytes.length)
           val os = httpExchange.getResponseBody
           try {
             os.write(bytes)
-          }
-          finally
-            os.close()
-        }
-        else {
-          httpExchange.sendResponseHeaders(404, -1)
-        }
+            if (shouldUseCompression(httpExchange)) {
+              val gzip = new GZIPOutputStream(os)
```

Comment:
Looking at the example code you referenced, you should set the content encoding header:
```
httpExchange.getResponseHeaders.set("Content-Encoding", "gzip")
```

Reply:
Added in https://github.com/kamon-io/Kamon/pull/915/commits/d98f3506d36dbfb309334a1bfe3f61622dca6dc0.